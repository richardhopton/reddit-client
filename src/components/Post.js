import React, { Component } from "react";
import PropTypes from "prop-types";
import Thumbnail from "./Thumbnail";
import styles from "./styles.css";

export default class Post extends Component {
  render() {
    const {
      score,
      thumbnail,
      author,
      permalink: url,
      title,
      num_comments: commentCount
    } = this.props;
    return (
      <div className={styles.post}>
        <Thumbnail source={thumbnail} />
        <div className={styles.postText}>
          <div className={styles.author}>
            Submitted by:{" "}
            <a href={`//www.reddit.com/user/${author}`}>{author}</a>
          </div>

          <div className={styles.score}>Score: {score}</div>

          <div className={styles.title}>
            <a target="_blank" href={`//www.reddit.com${url}`}>
              {title}
            </a>
          </div>

          <div className={styles.comments}>
            <a target="_blank" href={`//www.reddit.com${url}`}>
              {`${commentCount} comments`}
            </a>
          </div>
        </div>
      </div>
    );
  }
}

Post.propTypes = {
  title: PropTypes.string.isRequired,
  score: PropTypes.number.isRequired,
  thumbnail: PropTypes.string,
  author: PropTypes.string.isRequired,
  num_comments: PropTypes.number.isRequired,
  permalink: PropTypes.string.isRequired
};
