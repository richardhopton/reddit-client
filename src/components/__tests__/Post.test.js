import React from "react";
import Post from "../Post";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

const props = {
  id: 1,
  title: "1",
  name: "post1",
  score: 1,
  author: "user1",
  num_comments: 1,
  permalink: "http://1"
};

// Check the component hasn't been accidentally adjusted by using a jest snapshot
test("renders expected snapshot", () => {
  const tree = renderer.create(<Post {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
