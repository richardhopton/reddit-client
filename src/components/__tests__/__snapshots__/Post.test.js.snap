// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders expected snapshot 1`] = `
<div
  className="post"
>
  <div
    className="thumbnail"
  >
    <img
      className="thumbnailImage"
      onError={[Function]}
    />
  </div>
  <div
    className="postText"
  >
    <div
      className="author"
    >
      Submitted by:
       
      <a
        href="//www.reddit.com/user/user1"
      >
        user1
      </a>
    </div>
    <div
      className="score"
    >
      Score: 
      1
    </div>
    <div
      className="title"
    >
      <a
        href="//www.reddit.comhttp://1"
        target="_blank"
      >
        1
      </a>
    </div>
    <div
      className="comments"
    >
      <a
        href="//www.reddit.comhttp://1"
        target="_blank"
      >
        1 comments
      </a>
    </div>
  </div>
</div>
`;
